mne-demo/plot_sleep_batch.py

```diff
@@ -0,0 +1,129 @@
+import numpy as np
+import matplotlib.pyplot as plt
+
+import mne
+from mne.datasets.sleep_physionet.age import fetch_data
+from mne.time_frequency import psd_array_welch
+
+from sklearn.ensemble import RandomForestClassifier
+from sklearn.metrics import accuracy_score
+from sklearn.metrics import confusion_matrix
+from sklearn.metrics import classification_report
+from sklearn.pipeline import make_pipeline
+from sklearn.preprocessing import FunctionTransformer
+from sklearn.model_selection import train_test_split
+
+ALL_SUBJECTS = list(range(20))
+
+mapping = {'EOG horizontal': 'eog',
+           'Resp oro-nasal': 'misc',
+           'EMG submental': 'misc',
+           'Temp rectal': 'misc',
+           'Event marker': 'misc'}
+
+# With ALL data
+all_files_first_night = fetch_data(subjects=ALL_SUBJECTS, recording=[1])
+all_raw_edf = []
+
+for file in all_files_first_night:
+    raw = mne.io.read_raw_edf(file[0])
+    annot = mne.read_annotations(file[1])
+
+    raw.set_annotations(annot, emit_warning=False)
+    raw.set_channel_types(mapping)
+
+    all_raw_edf.append(raw)
+
+annotation_desc_2_event_id = {'Sleep stage W': 1,
+                              'Sleep stage 1': 2,
+                              'Sleep stage 2': 3,
+                              'Sleep stage 3': 4,
+                              'Sleep stage 4': 4,
+                              'Sleep stage R': 5}
+
+all_events = []
+for raw_sample in all_raw_edf:
+    events, _ = mne.events_from_annotations(raw_sample, event_id=annotation_desc_2_event_id, chunk_duration=30.)
```

Comment:
I see 30 is being used often. Have you tried different values for this? Would we want to make it a variable for easy tweaking?

Reply:
Ah yeah, I can go ahead and make these functions a bit more modular, I'll ping you when I push the changes

Reply:
I think it's because each event is 30 sec long